### PR TITLE
Add coverage tooling, update test tooling and improve coverage

### PR DESCRIPTION
**Pull request by no1seman**
0. Update luatest, luacheck module versions
1. Add coverage tooling
2. Add Makefile
3. Minor coverage Improvements

### CODE REVIEW

File: .luacov [hunk 1]
```diff
@@ -0,0 +1,6 @@
+statsfile = 'tmp/luacov.stats.out'
+reportfile = 'tmp/luacov.report.out'
+exclude = {
+    '/test/',
+    '/tmp/',
+}
```
Comment (olegrok) [inline .luacov]: Git warns about missing empty line

File: .gitignore [hunk 2]
```diff
@@ -27,4 +28,5 @@ luacov.*.out*
 /package-lock.json
 *.mo
 .history
-.vscode
+*.rock
+
```
Comment (olegrok) [inline .gitignore]: Extra empty line

File: graphql/execute.lua [hunk 3]
```diff
@@ -167,7 +167,7 @@ local function collectFields(objectType, selections, visitedFragments, result, c
 end
 
 local evaluateSelections
-local serializemap = {__serialize='map'}
+local serializemap = {__serialize='map',}
```
Comment (olegrok) [inline graphql/execute.lua]: It makes code uglier and I suggest to discard such changes.

**Review by olegrok — COMMENTED** [review]
Thanks for your patch. See my comments below.

Also commit message is empty. Please add info about your changes.

**Comment by Totktonada** [comment]
It looks relevant to https://github.com/keplerproject/luacov/issues/52, but this problem is not mentioned anywhere.

**Comment by olegrok** [comment]
> It looks relevant to [keplerproject/luacov#52](https://github.com/keplerproject/luacov/issues/52), but this problem is not mentioned anywhere.

Personally, I'm not sure that it's significant reason for making code uglier.

**Comment by DifferentialOrange** [comment]
I was finishing this PR as a part of this sprint tasks. You can see reworked version in #24 . Since this PR is based on branch of no1seman/graphql which I have no access to, I have opened a new PR.

**Comment by DifferentialOrange** [comment]
Closed by #24

**Pull request closed without merging**